### PR TITLE
Additional changes needed to compile on swift 1.2

### PR DESCRIPTION
**Pull request by nathankot**
Backwards breaking:
Can no longer do variadic @autoclosure arguments,
@see http://stackoverflow.com/questions/29750244/variadic-autoclosures-in-swift-1-2

### CODE REVIEW

File: Swiftz/OptionalExt.swift [hunk 1]
```diff
@@ -49,13 +49,13 @@ public func coalesce<T>(all : @autoclosure () -> T? ...) -> T? {
 
 /// Given one or more Optional values, returns the first Optional value that is not nil
 /// and satisfies the approve function when evaulated from left to right
-public func coalesce<T>(approve : T -> Bool) -> (@autoclosure () -> T? ...) -> T? {
-	return { all in
+public func coalesce<T>
```
Comment (CodaFi) [inline Swiftz/OptionalExt.swift]: I fear we may have to remove these functions altogether (see http://stackoverflow.com/a/29750368/945847).

Comment (nathankot) [inline Swiftz/OptionalExt.swift]: I leave that up to you :) yeah there's no way to have implicit laziness without autoclosure...

Comment (CodaFi) [inline Swiftz/OptionalExt.swift]: Let's compromise.  Comment these out and leave something above the function comments like 

```
/// Forbidden by Swift 1.2; see ~( http://stackoverflow.com/a/29750368/945847 )
```

Comment (nathankot) [inline Swiftz/OptionalExt.swift]: Done, see push :)

**Comment by CodaFi** [comment]
Thanks for putting up with me!

:boat:

**Pull request merged**